tidy3d/components/source.py

```diff
@@ -608,9 +608,42 @@ class GaussianBeam(AngledFieldSource, PlanarSource):
     )
 
 
+class AstigmaticGaussianBeam(AngledFieldSource, PlanarSource):
+    """Astigmatic Guassian distribution on finite extent plane.
```

Comment:
I feel like "astigmatic gaussian beam" is not a very well known thing. It was a surprise to me that you can have separate waist distance along x and y, apart from just different waist raidus. I guess this is what makes it astigmatic actually, as opposed to just elliptical.

Should we add the reference to the specific paper - or a little more epxlanation?

Reply:
Good idea, I added an explanation and a reference.